lib/build/recipe/thirdpartypasswordless/types.d.ts

```diff
@@ -0,0 +1,263 @@
+            preAuthSessionId: string;
+        } & {
+            config: PasswordlessConfig;
+        }
+    ) => Promise<
+        | {
+              status: "OK" | "RESTART_FLOW_ERROR";
+          }
+        | {
+              status: "GENERAL_ERROR";
+              message: string;
+          }
+    >;
+    consumeCode: (
+        input: (
+            | {
+                  userInputCode: string;
+                  deviceId: string;
+                  preAuthSessionId: string;
+              }
+            | {
+                  preAuthSessionId: string;
+                  linkCode: string;
+              }
+        ) & {
+            config: PasswordlessConfig;
+        }
+    ) => Promise<
+        | {
+              status: "OK";
+              createdUser: boolean;
+              user: PasswordlessUser;
+          }
+        | {
+              status: "INCORRECT_USER_INPUT_CODE_ERROR" | "EXPIRED_USER_INPUT_CODE_ERROR";
+              failedCodeInputAttemptCount: number;
+              maximumCodeInputAttempts: number;
+          }
+        | {
+              status: "GENERAL_ERROR";
+              message: string;
+          }
+        | {
+              status: "RESTART_FLOW_ERROR";
+          }
+    >;
+    doesEmailExist: (input: { email: string; config: PasswordlessConfig }) => Promise<boolean>;
+    doesPhoneNumberExist: (input: { phoneNumber: string; config: PasswordlessConfig }) => Promise<boolean>;
+    getLoginAttemptInfo: () =>
+        | Promise<
+              | undefined
+              | {
+                    deviceId: string;
+                    preAuthSessionId: string;
+                    contactInfo: string;
+                    contactMethod: "EMAIL" | "PHONE";
+                    flowType: "USER_INPUT_CODE" | "MAGIC_LINK" | "USER_INPUT_CODE_AND_MAGIC_LINK";
+                    lastResend: number;
+                    redirectToPath?: string;
+                }
+          >
+        | {
+              deviceId: string;
+              preAuthSessionId: string;
+              contactInfo: string;
+              contactMethod: "EMAIL" | "PHONE";
+              flowType: "USER_INPUT_CODE" | "MAGIC_LINK" | "USER_INPUT_CODE_AND_MAGIC_LINK";
+              lastResend: number;
+              redirectToPath?: string;
+          }
+        | undefined;
+    setLoginAttemptInfo: (input: {
+        deviceId: string;
+        preAuthSessionId: string;
+        contactInfo: string;
+        contactMethod: "EMAIL" | "PHONE";
+        flowType: "USER_INPUT_CODE" | "MAGIC_LINK" | "USER_INPUT_CODE_AND_MAGIC_LINK";
+        lastResend: number;
+        redirectToPath?: string;
+    }) => Promise<void> | void;
+    clearLoginAttemptInfo: () => Promise<void> | void;
+    getOAuthAuthorisationURL: (input: { thirdPartyId: string; config: TPConfig }) => Promise<string>;
+    signInAndUp: (input: { thirdPartyId: string; config: TPConfig }) => Promise<
+        | {
+              status: "OK";
+              user: User;
```

Comment:
This `User` type is not correct. It should be the same type as that in passwordless recipe (since that is a superset of third party user)

Reply:
I'm not sure. Why are we returning the superset when it's always the more specific type? That way devs using this would have to assert that the email prop is present on the returned value.

Reply:
Email may not be present if a user used passwordless via mobile. Hence it needs to be optional

Reply:
@porcellus did you miss this?

Reply:
In a recipe, we use just one user type across all of its functions. In this recipe, the user type should have both email and phone number as optional, therefore, this function must return that user type.

Reply:
One problem is that we need to assign this to the signInAndUp of the third-party recipe, which requires this user type. Other than that, maybe `PasswordlessUser | User` would be the most precise because that is what the user receives in the `SUCCESS` event.

Reply:
Let's discuss this on a call.